pkg/cri/config/config_kernel_linux_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"testing"
+
+	kernel "github.com/containerd/containerd/contrib/seccomp/kernelversion"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateEnableUnprivileged(t *testing.T) {
+	expectedErrFunc := func() string {
+		fourDotEleven := kernel.KernelVersion{Kernel: 4, Major: 11}
+		ok, _ := kernel.GreaterEqualThan(fourDotEleven)
+		if !ok {
+			return "unprivileged_icmp and unprivileged_port require kernel version greater than or equal to 4.11"
+		}
+		return ""
+	}
```

Comment:
This doesn't really test anything since the logic for "expected error" is the same kernel comparison that the code-under-test is doing.

Another approach would be replacing the function that's being called.  For example (not necessarily this exact code, but a possible approach):

`config_kernel_linux.go`
```
var kernelGreaterEqualThan = kernel.GreaterEqualThan
```

`config_kernel_linux_test.go`
```
origKernelGreaterEqualThan := kernelGreaterEqualThan
t.Cleanup(func () {
	kernelGreaterEqualThan = origKernelGreaterEqualThan
})
kernelGreaterEqualThan = func(minVersion KernelVersion) (bool, err) {
	return test.kernelGreater, nil	
}
```

Reply:
This is great advice, thank you.